src/cli/repl/core.ts

```diff
@@ -21,33 +21,90 @@ const replCompleterKeywords = Array.from(commandNames, s => `:${s}`)
  * Used by the repl to provide automatic completions for a given (partial) input line
  */
 export function replCompleter(line: string): [string[], string] {
-	const singleCommand = replCompleterKeywords.find(k => (k + ' ') === line)
-	if(singleCommand !== undefined){
-		const applicableCommand = getCommand(singleCommand.substring(1))
-		guard(applicableCommand !== undefined, 'Command should be defined')
-		if(applicableCommand.script){
-			let scriptOptions: OptionDefinition[] | undefined
-			for(const [script, { options}] of Object.entries(scripts)){
-				if(script === line.trimEnd().substring(1)){
-					scriptOptions = options
-					break
-				}
-			}
-			guard(scriptOptions !== undefined, 'script should be in script record')
-			const autoCompleteList : string[] = []
-			autoCompleteList.push(line)
-			scriptOptions.forEach(option => {
-				autoCompleteList.push(line + ' --' + option.name)
-				if(option.alias !== undefined){
-					autoCompleteList.push(line + ' -' + option.alias)
+	const splitCommandList = splitAtEscapeSensitive(line)
+	console.log(splitCommandList)
+	//find command in commandlist
+	const keyword = replCompleterKeywords.find(k => splitCommandList[0] === k)
+	if(keyword !== undefined && keyword.startsWith(':')){
+		const singleCommand = getCommand(keyword.slice(1))
+		guard(singleCommand !== undefined, 'Keyword does not match any command')
+		if(singleCommand.script){
+			const scriptInformation = getScriptInformation(keyword.slice(1))
+			guard(scriptInformation !== undefined, 'script should be in script record')
+			const scriptOptions: OptionDefinition[] = scriptInformation.options
+			//console.log(scriptOptions)//TODO REMOVE
+			const possibleOptions : string[] = []
+			getPossibleCommandLineOptions(scriptOptions,splitCommandList).forEach(o => {
+				possibleOptions.push('--' + o.name)
```

Comment:
it feels like we should abstract away from the constant `--` and `-` prefix-addition.